lib/bolt/pal/yaml_plan/step.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+require 'bolt/util'
+
+module Bolt
+  class PAL
+    class YamlPlan
+      class Step
+        attr_reader :name, :type, :body, :target
+
+        COMMON_STEP_KEYS = %w[name description target].freeze
+        STEP_KEYS = {
+          'command' => {
+            'allowed_keys' => Set['command'].merge(COMMON_STEP_KEYS),
+            'required_keys' => Set['target']
+          },
+          'script' => {
+            'allowed_keys' => Set['script', 'parameters', 'arguments'].merge(COMMON_STEP_KEYS),
+            'required_keys' => Set['target']
+          },
+          'task' => {
+            'allowed_keys' => Set['task', 'parameters'].merge(COMMON_STEP_KEYS),
+            'required_keys' => Set['target']
+          },
+          'plan' => {
+            'allowed_keys' => Set['plan', 'parameters'].merge(COMMON_STEP_KEYS),
+            'required_keys' => Set.new
+          },
+          'source' => {
+            'allowed_keys' => Set['source', 'destination'].merge(COMMON_STEP_KEYS),
+            'required_keys' => Set['target', 'source', 'destination']
+          },
+          'destination' => {
+            'allowed_keys' => Set['source', 'destination'].merge(COMMON_STEP_KEYS),
+            'required_keys' => Set['target', 'source', 'destination']
+          },
+          'eval' => {
+            'allowed_keys' => Set['eval', 'name', 'description'],
+            'required_keys' => Set.new
+          }
+        }.freeze
+
+        def initialize(step_body, step_number)
+          @body = step_body
+          @name = @body['name']
+          # For error messages
+          @step_number = step_number
+          validate_step
+
+          @type = STEP_KEYS.keys.find { |key| @body.key?(key) }
+          @target = @body['target']
+        end
+
+        def transpile(plan_path)
+          result = String.new("  ")
+          result << "$#{@name} = " if @name
+
+          description = body.fetch('description', nil)
+          parameters = body.fetch('parameters', {})
+          if @type == 'script' && body.key?('arguments')
+            parameters['arguments'] = body['arguments']
+          end
+
+          case @type
+          when 'command', 'task', 'script', 'plan'
+            result << "run_#{@type}(#{Bolt::Util.to_code(body[@type])}"
+            result << ", #{Bolt::Util.to_code(@target)}" if @target
+            result << ", #{Bolt::Util.to_code(description)}" if description
```

Comment:
I dont think `run_plan` accepts a description. 
```
steps:
  - name: run_plan
    plan: sample::single_task
    parameters:
      nodes: localhost
    description: foo

return: run_plan
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt plan convert site/yaml_plans/plans/nested/nested_yaml.yaml
# WARNING: This is an autogenerated plan. It may not behave as expected.
plan yaml_plans::nested::nested_yaml() {
  $run_plan = run_plan('sample::single_task', 'foo', {'nodes' => 'localhost'})

  return 'run_plan'
}
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt plan run yaml_plans::nested::nested_yaml_pp
Starting: plan yaml_plans::nested::nested_yaml_pp
Finished: plan yaml_plans::nested::nested_yaml_pp in 0.0 sec
{
  "kind": "bolt/pal-error",
  "msg": "'run_plan' expects between 1 and 2 arguments, got 3",
  "details": {
  }
}
```